### DISCUSSION TITLE
Definition of capacity

### DISCUSSION THREAD

**Discussion by harisbal** [jdlph/Path4GMNS#52] [Q&A] [answered]
Hello, and thank you very much for this very useful library!

I am a bit confused about the value of capacity we should include in the `link.csv' file.
Based on what I see in the code, I believe that the capacity in the file should be the total capacity of the link (capacity * lanes * assignment duration):
https://github.com/jdlph/Path4GMNS/blob/fe8ab94b77d625d463714ce5b0f39e5f10d07103/path4gmns/io.py#L184C5-L187C5

However, GMNS specifies capacity as per lane per hour:
https://github.com/zephyr-data-specs/GMNS/blob/main/spec/link.schema.json#L136-L143  

How should we specify capacity in path4gmns?

**Comment by jdlph** [comment, marked as answer]
The following line of code from the constructor of Class Link shall explain. https://github.com/jdlph/Path4GMNS/blob/fe8ab94b77d625d463714ce5b0f39e5f10d07103/path4gmns/classes.py#L98

This Q&A is marked as closed. You can reopen it if you have related questions.

**Reply by harisbal** [reply]
I have a follow up question actually.
Does the duration of the time period influence the capacity of the link?

For example, if the assigned time period is 2 hours (e.g. from 0700 to 0900) then do we need to manually double the capacity of each link or not?

**Comment by harisbal** [comment]
Thank you for the clarification!
Cheers
